### ISSUE TITLE
Read the Ripples fails with NPCs in the room

### ISSUE THREAD

**Issue by mwest152** [rpherbig/dr-scripts#1904]
Someone did some detective work and found the check is in buff.lic I think 'return if DRRoom.npcs.empty

Not sure if it checks the ignored_npc: list or not.  I tried adding the NPCs to that list and it didnt seem to work.

Specifically with RTR, if you stand while its up it falls off.  So retreating because of NPCs causes you to stand and try to retreat.

Happens with all rituals around NPCs as far as I can tell, but this one specifically you can't use while standing.

LOG BELOW:
-------------------------------

```
[astrology]>put bones in my pack

You put your bones in your grey pack.
> 
The brilliant silver glow dissipates from Lucelle.
> 
[astrology]>retreat

You are already as far away as you can get!
> 
[astrology]>stance set 100 0 80

Setting your Evasion stance to 100%, your Parry stance to 0%, and your Shield stance to 80%.  You have 10 stance points left.
> 
[astrology]>prepare RtR 400

You feel intense strain as you try to manipulate the mana streams to form this pattern, and you are not certain that you will have enough mental stamina to complete it.
However, if you utilize a ritual focus: That won't affect your current attunement very much.

You raise your head skyward, chanting the equation of the Read the Ripples spell.
> 
[astrology]>get my abacus

> 
You get a moonstone-beaded watered steel abacus etched with complex sigils from inside your grey pack.
> 
[astrology]>invoke my abacus 

You lift your abacus toward the sky and will the mana streams to bend into it.
Roundtime: 20 sec.
R> 

Your abacus steadily brightens, until it burns like a glowing white torch!
You direct the flow of power you've conjured up into your spell pattern, taking much of the strain of empowering it off you.
R> 

Your abacus fades over the course of a few seconds, returning to normal.
R> 

[astrology]>stow my abacus

You put your abacus in your grey pack.
> 
[astrology]>retreat

You are already as far away as you can get!
> 
[astrology]>sit

> 
You feel fully prepared to cast your spell.
> 
You sit down.

s> 
[astrology]>cast

You gaze towards the heavens seeking their silent guidance.
The mental strain of this pattern is considerably eased by your ritual focus.
At the ritual's peak, your prophetic connection blooms a thousand-fold.  You are alone.  An infinitesimal speck in space and time adrift in an infinite sea of possibility.  The course of your past, present and future are dictated by ceaseless currents beyond any mortal control.

But infinitesimal is not insignificant, and control is not understanding.  These are indisputable facts etched into the essence of every prophet.

s> 
[astrology]>retreat

You must stand first.
s> 
[astrology]>stand

You stand back up.

> 
[astrology]>retreat

You are already as far away as you can get!
> 
You feel fully rested.
You feel fully attuned to the mana streams again.

> 
You find it impossible to maintain your ritual while standing.
As your ritual ebbs away, you find yourself returned to a world of immutable, indisputable facts.  Things that are, are, and things that are not, are not.  A faint dissatisfaction lingers with you, for you are still alone, infinitesimal.
```

**Comment by jandersson** [comment]
I checked the method `common-travel::retreat` and it does not check `ignored_npcs`

**Comment by mwest152** [comment]
an astrology_room: setting would be a decent work around for me if that's an easy one.  Can just make sure to use a room that is regularly empty.

**Comment by jandersson** [comment]
Whats a good solution? Check for ignored npcs? Or maybe distinguish combat vs noncombat rituals

**Comment by jandersson** [comment]
You will never cast RtR in combat, so thats a plus for distinguishing noncombat vs combat rituals. I'm sure theres other spells that fall into this case

**Comment by mwest152** [comment]
Sounds like ignored_npcs is the easy universal fix,  unless there is a good reason to adjust them all.

**Comment by rpherbig** [comment]
Fixed by #1976: acf5a30d5fc9a05d06192dfa11e4c614988a30a2
